smithy-aws-protocol-tests/model/restJson1/malformedRequests/malformed-string.smithy

```diff
@@ -0,0 +1,59 @@
+$version: "1.0"
+
+namespace aws.protocoltests.restjson
+
+use aws.protocols#restJson1
+use smithy.test#httpMalformedRequestTests
+
+@http(uri: "/MalformedString/{pleaseNo}", method: "POST")
+operation MalformedString {
+    input: MalformedStringInput
+}
+
+apply MalformedString @httpMalformedRequestTests([
+    {
+        id: "RestJsonHeaderMalformedStringInvalidBase64MediaType",
+        documentation: """
+        When string with the mediaType trait is bound to a header, its value
+        must be base64 encoded. The server should reject values that aren't
+        valid base64 out of hand.""",
+        protocol: restJson1,
+        request: {
+            method: "POST",
+            uri: "/MalformedString",
+            body: """
+            { "blob" : $value:L }""",
+            headers: {
+                "content-type": "application/json",
+                "amz-media-typed-header": "$value:L",
+            }
+        },
+        response: {
+            code: 400,
+            headers: {
+                "x-amzn-errortype": "SerializationException"
+            }
+        },
+        testParameters: {
+            value: [
+                // Insufficient padding
+                "xyz",
+                // Extant, but also insufficient padding
+                "YmxvYg=",
+                // Invalid characters
+                "[][]",
+                // Invalid characters which are commonly used as filename-safe
+                // alternatives to + and /
+                "-_=="
```

Comment:
Do we want to accept these or not? They're often used for uri-safe and filename-safe encoding. They can also be percent encoded in the uri case.

This also raises the question of whether media-typed strings should be encoded when used in a URI. Percent encoding should be sufficient there, I think.

Reply:
Yeah, percent encoding sounds right to me.